katdal/chunkstore.py

```diff
@@ -316,20 +335,26 @@ def put_dask_array(self, array_name, array):
         array : :class:`dask.array.Array` object
             Dask input array
 
-        Raises
-        ------
-        :exc:`chunkstore.StoreUnavailable`
-            If interaction with chunk store failed (offline, bad auth, bad config)
-        :exc:`chunkstore.ChunkNotFound`
-            If `array_name` is incompatible with store
+        Returns
+        -------
+        success : :class:`dask.array.Array` object
+            Dask array indicating success of the transfer of each chunk
+            (None indicates success, otherwise there is an exception object)
         """
         dask_graph = dask.sharedict.ShareDict()
         dask_graph.update(array.dask)
+        # Give better names to these two very similar variables
+        in_name = array.name
+        out_name = array_name
+        # Disambiguate out_name to avoid clashes in the dask graph
+        if out_name == in_name:
+            out_name = 'store-' + out_name
```

Comment:
There is no guarantee that this is globally unique. For example, if you write an array into two different chunk stores with the same array name, these will clash. Writing to a chunk store isn't really a "pure" function (just as one example, it may return an exception the first time, and succeed the second time), so I'd suggest using 'store-' + array_name + random string (I forget how to get one, but I know dask has the capability).

Reply:
Of course. I just followed the lead of the dask gurus in their equivalent `da.core.insert_to_ooc` function.

I'm also considering a unique chunk store ID for this purpose, to make debugging easier. The `da.core.tokenize` function can give you a deterministic hash of the unique parts of the store.

Reply:
Made this mostly unique now with a random token (1 in 2 ** 32 is good enough for me).

Reply:
Dask seems to use `str(uuid.uuid1())`, which should give better than 1 in 2<sup>32</sup>.

Reply:
Yes, but those long strings hurt my eyes :-)